### PR TITLE
🅰️ feat: Dynamic Font Size

### LINKED ISSUE #3207: Enhancement: Font & Font Size option
### What features would you like to see added?

I have some eyesight issues and I'd like the ability to set the font size of the interface.

If possible, I'd also like the ability to set the interface font (I have some more readable fonts I like to use). That might be a bit difficult, but atleast setting the font size would be chat!

### More details

Be great to have a section in user settings somewhere.

### Which components are impacted by your request?

_No response_

### Pictures

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by danny-avila**
### Summary

closes https://github.com/danny-avila/LibreChat/issues/3207

I made several improvements to the LibreChat application to enhance user experience and readability, focusing on styling updates and the addition of a dynamic font size feature.

- Updated `HoverButtons` styling for better visibility and user interaction.
- Moved `FontSizeSelector` component from General settings to Chat settings.
- Introduced dynamic font size adjustments in `ProgressText` and `MessageContent` components.
- Refined code analyze block styling for consistency.
- Modified message font size settings in the navigation bar.
- Implemented dynamic font size functionality across various components, including `MessageParts`, `MessagesView`, `SearchMessage`, and more.
- Updated `PromptForm` component to make the `selectedPrompt` type optional.
- Adjusted `IntersectionObserver` threshold and debounce rate for improved performance.
- Refactored `FontSizeSelector` options to enhance the user interface.
- Added localization for font size options in multiple languages.

### Change Type
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

### Testing
- Tested the visibility and interaction of `HoverButtons` to ensure they are more user-friendly.
- Verified the correct relocation and functionality of the `FontSizeSelector` component.
- Checked dynamic font size adjustments in `ProgressText` and `MessageContent` for different font sizes.
- Ensured that code analyze block styling is consistent and readable.
- Validated dynamic font size changes across multiple components, including `MessageParts` and `MessagesView`.
- Updated unit tests to reflect the `selectedPrompt` type changes in `PromptForm`.
- Measured performance improvements with the adjusted `IntersectionObserver` threshold and debounce rate.
- Reviewed localization updates for accuracy in multiple languages.

### Checklist
- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes

### CODE REVIEW

File: client/src/style.css [hunk 1]
```diff
@@ -2184,3 +2204,16 @@ ol ol):not(:where([class~=not-prose] *)) {
   animation: moveUp 4s ease-in-out infinite;
   
 }
+
+.message-content {
+  font-size: var(--markdown-font-size, 1rem);
```
Comment (smadep) [inline client/src/style.css:2209]: shouldn't that fallback "1rem" also be var(--font-size-base)?

File: client/src/style.css [hunk 2]
```diff
@@ -2184,3 +2204,16 @@ ol ol):not(:where([class~=not-prose] *)) {
...
+  line-height: 1.75;
+}
+
+.message-content pre code {
+  font-size: calc(0.85 * var(--markdown-font-size, 1rem));
```
Comment (smadep) [inline client/src/style.css:2214]: shouldn't that fallback "1rem" also be var(--font-size-base)?

File: client/src/style.css [hunk 3]
```diff
@@ -2184,3 +2204,16 @@ ol ol):not(:where([class~=not-prose] *)) {
...
+}
+
+.message-content pre {
+  font-size: var(--markdown-font-size, 1rem);
```
Comment (smadep) [inline client/src/style.css:2218]: shouldn't that fallback "1rem" also be var(--font-size-base)?

**Comment by danny-avila** [comment]
@smadep thanks for your feedback, have you tried it? I'll test it out

**Comment by smadep** [comment]
Didn‘t do that yet, but I can try later. I‘m not familiar enough yet with the project to know right away where to test that style, just saw, that in similar cases the base variable was used as fallback.

I‘m in the process to add some similar UI setting to control the width of the chat-panel; on large displays (xl) the 48rem appear quite narrow to me. And I don‘t like to scroll the sample code response horizontally.And instead of a hardcoded change in my personal fork I thought I could contribute with a new setting?!

While I tried to understand how this is done for fontSize I found above.

**Comment by smadep** [comment]
@danny-avila, I changed style sheet for all occurances where --markdown-font-size was used with 1rem as fallback to use font-size-base ... seems to be working.

**Comment by danny-avila** [comment]
>  And I don‘t like to scroll the sample code response horizontally.And instead of a hardcoded change in my personal fork I thought I could contribute with a new setting?!

Absolutely, contributions always welcome. Something like this is aligned with what I would like improvements for.

> I changed style sheet for all occurances where --markdown-font-size was used with 1rem as fallback to use font-size-base ... seems to be working.

Thank you, I will make that change, I think I overlooked this.

**Comment by danny-avila** [comment]
@smadep made the change here, thanks!

https://github.com/danny-avila/LibreChat/pull/3600

**Pull request merged**